src/gui/processing/qgsprocessingmaplayercombobox.cpp

```diff
@@ -180,8 +187,19 @@ QgsProcessingMapLayerComboBox::QgsProcessingMapLayerComboBox( const QgsProcessin
 
   if ( filters )
     mCombo->setFilters( filters );
+
   mCombo->setExcludedProviders( QStringList() << QStringLiteral( "grass" ) ); // not sure if this is still required...
 
+  // see https://github.com/qgis/QGIS/issues/55890
+  // There is a chance that "qgis" algorithms can handle virtual rasters but other almost certainly cannot
+  // so we exclude them from the list of available layers for processing algorithms
+  const QStringList providersCompatibleWithVirtualRaster {{ QStringLiteral( "qgis" ), QStringLiteral( "native" ) }};
```

Comment:
instead of hardcoding this logic here, could we make it a virtual method in QgsProcessingProvider ? It could even be a new QgsProcessingProvider::Flag value.

Reply:
Done